pkg/metrics/queue.go

```diff
@@ -72,70 +92,89 @@ func InitQueueMetrics(name string) CoreQueueMetrics {
 }
 
 func (m *QueueMetrics) Reset() {
-	m.appMetrics.Reset()
-	m.ResourceMetrics.Reset()
+	m.appMetricsLabel.Reset()
+	m.appMetricsSubsystem.Reset()
+	m.ResourceMetricsLabel.Reset()
+	m.ResourceMetricsSubsystem.Reset()
 }
 
 func (m *QueueMetrics) IncQueueApplicationsRunning() {
-	m.appMetrics.With(prometheus.Labels{"state": "running"}).Inc()
+	m.appMetricsLabel.With(prometheus.Labels{"state": "running"}).Inc()
+	m.appMetricsSubsystem.With(prometheus.Labels{"state": "running"}).Inc()
 }
 
 func (m *QueueMetrics) DecQueueApplicationsRunning() {
-	m.appMetrics.With(prometheus.Labels{"state": "running"}).Dec()
+	m.appMetricsLabel.With(prometheus.Labels{"state": "running"}).Dec()
+	m.appMetricsSubsystem.With(prometheus.Labels{"state": "running"}).Dec()
+
 }
 
 func (m *QueueMetrics) IncQueueApplicationsAccepted() {
-	m.appMetrics.With(prometheus.Labels{"state": "accepted"}).Inc()
+	m.appMetricsLabel.With(prometheus.Labels{"state": "accepted"}).Inc()
+	m.appMetricsSubsystem.With(prometheus.Labels{"state": "accepted"}).Inc()
 }
 
 func (m *QueueMetrics) IncQueueApplicationsRejected() {
-	m.appMetrics.With(prometheus.Labels{"state": "rejected"}).Inc()
+	m.appMetricsLabel.With(prometheus.Labels{"state": "rejected"}).Inc()
+	m.appMetricsSubsystem.With(prometheus.Labels{"state": "rejected"}).Inc()
 }
 
 func (m *QueueMetrics) IncQueueApplicationsFailed() {
-	m.appMetrics.With(prometheus.Labels{"state": "failed"}).Inc()
+	m.appMetricsLabel.With(prometheus.Labels{"state": "failed"}).Inc()
+	m.appMetricsSubsystem.With(prometheus.Labels{"state": "failed"}).Inc()
 }
 
 func (m *QueueMetrics) IncQueueApplicationsCompleted() {
-	m.appMetrics.With(prometheus.Labels{"state": "completed"}).Inc()
+	m.appMetricsLabel.With(prometheus.Labels{"state": "completed"}).Inc()
+	m.appMetricsSubsystem.With(prometheus.Labels{"state": "completed"}).Inc()
 }
 
 func (m *QueueMetrics) IncAllocatedContainer() {
-	m.appMetrics.With(prometheus.Labels{"state": "allocated"}).Inc()
+	m.appMetricsLabel.With(prometheus.Labels{"state": "allocated"}).Inc()
+	m.appMetricsSubsystem.With(prometheus.Labels{"state": "allocated"}).Inc()
 }
 
 func (m *QueueMetrics) IncReleasedContainer() {
-	m.appMetrics.With(prometheus.Labels{"state": "released"}).Inc()
+	m.appMetricsLabel.With(prometheus.Labels{"state": "released"}).Inc()
+	m.appMetricsSubsystem.With(prometheus.Labels{"state": "released"}).Inc()
 }
 
 func (m *QueueMetrics) SetQueueGuaranteedResourceMetrics(resourceName string, value float64) {
-	m.ResourceMetrics.With(prometheus.Labels{"state": "guaranteed", "resource": resourceName}).Set(value)
+	m.ResourceMetricsLabel.With(prometheus.Labels{"state": "guaranteed", "resource": resourceName}).Set(value)
+	m.ResourceMetricsSubsystem.With(prometheus.Labels{"state": "guaranteed", "resource": resourceName}).Set(value)
```

Comment:
Same here, if `ResourceMetricsSubsystem` will be removed, add a comment in the struct.